server/models/procedures/invoicing.sql

```diff
@@ -749,14 +749,16 @@ BEGIN
     SELECT p.reference_uuid AS uuid, p.debit_equiv as debit, p.credit_equiv as credit, p.trans_date as date
     FROM posting_journal AS p
       JOIN cash c ON c.uuid = p.reference_uuid
-    WHERE c.debtor_uuid = debtor_uuid AND c.reversed = 0 AND c.is_caution = 1;
+    WHERE c.debtor_uuid = debtor_uuid AND c.reversed = 0 AND c.is_caution = 1
+      AND p.trans_id NOT IN (SELECT pj.trans_id FROM posting_journal pj JOIN invoice i ON i.uuid = pj.reference_uuid WHERE pj.trans_id = p.trans_id AND i.reversed = 1);
```

Comment:
For these, it's better to use `record_uuid`, since it is binary and faster to compare.

Reply:
You're right the query must be optimized because with real dataset there is a `ER_LOCK_WAIT_TIMEOUT'`

Reply:
@jniles the purpose of this procedure is to get a balance, so why not use a simple query for getting the balance of the debtor

Reply:
I'm pretty sure the purpose is to get the balance on this invoice, but I'll have to read the code to make sure.  I'll look into it.